src/main/java/com/ibm/cloud/sdk/core/util/DateUtils.java

```diff
@@ -13,80 +13,224 @@
 
 package com.ibm.cloud.sdk.core.util;
 
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
+import java.time.DateTimeException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.Arrays;
 import java.util.Date;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * This class contains utilities for formatting java.util.Date instances as a
- * "date" or a "date-time" value. In this context, "date" and "date-time" refer
- * to the data and datetime types that are part of the OpenAPI Specification.
+ * This class contains utilities for formatting and parsing {@link Date} instances as OpenAPI
+ * "date" or "date-time" values.
+ * <br><br>
+ * References:
+ * <dl>
+ * <dt>API Handbook:</dt>
+ * <dd>
+ * <ul>
+ * <li>https://test.cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date</li>
+ * <li>https://test.cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date-time</li>
```

Comment:
Since this is a public repo, we should link to the public API Handbook here.
```suggestion
 * <li>https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date</li>
 * <li>https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date-time</li>
```